Fix laravel transaction naming (issue #86) - Don't use getName() if it's a auto generated route name

stockalexander opened the pull request:
Route::getName() returns a random generated string (generated::*) if a route isn't named [(Named Routes)](https://laravel.com/docs/8.x/routing#named-routes) and routes are cached via `php artisan route:cache` since laravel 7.x.

This causes the problem mentioned in #86. To fix the problem, I added a check if the returned string begins with `generated::`, to skip it and use `Route::getAction()` instead.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/newrelic-php-agent?pullRequest=174) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/newrelic-php-agent?pullRequest=174) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/newrelic-php-agent?pullRequest=174) it.</sub>

Fahmy-Mohammed commented:
ok jenkins

Fahmy-Mohammed reviewed (dismissed):
Thanks for adding this fix @stockalexander. I've tested this locally and it works well! This should be included in the upcoming PHP agent release (version 9.18).

Fahmy-Mohammed commented:
@stockalexander before we can merge this, would you be able to change this to the dev branch instead of main? That is where all of the work included in the upcoming release is contained.

stockalexander commented:
@Fahmy-Mohammed: Nice! I rebased my branch to the dev branch (+ force pushed it) and updated the PR. Can you merge it now or should I change something else?

Fahmy-Mohammed commented:
@stockalexander There are some internal Jenkins builds that are running currently, so I'll merge this once those tests pass. Awesome work identifying and adding the fix. We really appreciate it!

Fahmy-Mohammed commented:
ok jenkins

Fahmy-Mohammed commented:
ok jenkins
